docker/runtime-entrypoint.sh

```diff
@@ -1,82 +1,95 @@
+: "${ZEBRA_CONF_FILE:='zebrad.toml'}"
+if [[ -n "$ZEBRA_CONF_DIR" ]] && [[ -n "$ZEBRA_CONF_FILE" ]]; then
+    ZEBRA_CONF_PATH="$ZEBRA_CONF_DIR/$ZEBRA_CONF_FILE"
+fi
 
-echo "Other variables:"
-echo "SHORT_SHA=$SHORT_SHA"
-echo "SENTRY_DSN=$SENTRY_DSN"
+# [network]
+: "${NETWORK:='Mainnet'}"
+: "${ZEBRA_LISTEN_ADDR:='127.0.0.1'}"
+# [consensus]
+: "${ZEBRA_CHECKPOINT_SYNC:='true'}"
+# [state]
+: "${ZEBRA_CACHED_STATE_DIR:='/var/cache/zebrad-cache'}"
+: "${LOG_COLOR:='false'}"
+# [rpc]
+: "${RPC_LISTEN_ADDR:='127.0.0.1'}"
 
 # Create the conf path and file if it does not exist.
-mkdir -p "$ZEBRA_CONF_DIR"
-touch "$ZEBRA_CONF_PATH"
+if [[ -n "$ZEBRA_CONF_PATH" ]]; then
+    mkdir -p "$ZEBRA_CONF_DIR"
+    touch "$ZEBRA_CONF_FILE"
+fi
 
 # Populate `zebrad.toml` before starting zebrad, using the environmental
-# variables set by the Dockerfile.
+# variables set by the Dockerfile or the user.
 #
 # We disable most ports by default, so the default config is secure.
 # Users have to opt-in to additional functionality by setting environmental variables.
-#
-# TODO:
-#  - make `cache_dir`, `metrics.endpoint_addr`, and `tracing.endpoint_addr` into Docker arguments
-#  - add an $EXTRA_CONFIG or $REPLACEMENT_CONFIG environmental variable
+if [[ ! -f "$ZEBRA_CONF_PATH" ]]; then
 cat <<EOF > "$ZEBRA_CONF_PATH"
 [network]
 network = "$NETWORK"
-listen_addr = "0.0.0.0"
+listen_addr = $ZEBRA_LISTEN_ADDR:8233
+
+[consensus]
+checkpoint_sync = $ZEBRA_CHECKPOINT_SYNC
 
 [state]
-cache_dir = "$ZEBRA_CACHED_STATE_DIR"
+cache_dir = $ZEBRA_CACHED_STATE_DIR
+EOF
 
+if [[ -n "$METRICS_ENDPOINT_ADDR" ]]; then
+cat <<EOF >> "$ZEBRA_CONF_PATH"
 [metrics]
-#endpoint_addr = "0.0.0.0:9999"
+endpoint_addr = ${METRICS_ENDPOINT_ADDR}:9999
 EOF
+fi
 
+# Set this to enable the RPC port
 if [[ -n "$RPC_PORT" ]]; then
 cat <<EOF >> "$ZEBRA_CONF_PATH"
 [rpc]
-listen_addr = "0.0.0.0:${RPC_PORT}"
+listen_addr = ${RPC_LISTEN_ADDR}:${RPC_PORT}
 EOF
 fi
 
-if [[ -n "$LOG_FILE" ]] || [[ -n "$LOG_COLOR" ]]; then
+if [[ -n "$TRACING_ENDPOINT_ADDR" ]]; then
 cat <<EOF >> "$ZEBRA_CONF_PATH"
 [tracing]
-#endpoint_addr = "0.0.0.0:3000"
+endpoint_addr = "${TRACING_ENDPOINT_ADDR}:3000"
 EOF
 fi
 
+# Set this to log to a file, if not set, logs to standard output
 if [[ -n "$LOG_FILE" ]]; then
-mkdir -p $(dirname "$LOG_FILE")
+mkdir -p "$(dirname "$LOG_FILE")"
 
 cat <<EOF >> "$ZEBRA_CONF_PATH"
 log_file = "${LOG_FILE}"
 EOF
 fi
 
-if [[ "$LOG_COLOR" = "true" ]]; then
+# Zebra automatically detects if it is attached to a terminal, and uses colored output.
+# Set this to 'true' to force using color even if the output is not a terminal.
+# Set this to 'false' to disable using color even if the output is a terminal.
+if [[ "$LOG_COLOR" = "true" ]] || [[ "$LOG_COLOR" = "false" ]]; then
 cat <<EOF >> "$ZEBRA_CONF_PATH"
-force_use_color = true
-EOF
-elif [[ "$LOG_COLOR" = "false" ]]; then
-cat <<EOF >> "$ZEBRA_CONF_PATH"
-use_color = false
+force_use_color = $LOG_COLOR
 EOF
```

Comment:
The previous script set `force_use_color` if the env var was true, and `use_color` if it was false. This is the correct behaviour, changing it is a breaking change.

Currently in this PR there's no way to force-disable colours.

Reply:
Change reverted